### PR TITLE
bindinfo: enforce default_db to lower case for SQL bind operations (#21861)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #21861 to release-4.0

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/21841

Problem Summary:

DROP BINDING does not take effect when default_db does not match from case sensitivity perspective.

### What is changed and how it works?

What's Changed:

- unify all the `default_db` to lower case for all SQL bind operations;
- for existing bindings, add `LOWER()` function in SQL to handle them;

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

N/A

### Release note 

- enforce default_db to lower case for SQL bind operations

### CODE REVIEW

File: bindinfo/bind_test.go [hunk 1]
```diff
@@ -740,6 +740,116 @@ func (s *testSuite) TestCapturePlanBaseline(c *C) {
 	c.Assert(rows[0][1], Equals, "SELECT /*+ use_index(@`sel_1` `test`.`t` )*/ * FROM `t` WHERE `a`>10")
 }
 
+<<<<<<< HEAD
```
Comment (qw4990) [inline bindinfo/bind_test.go:743]: PTAL @eurekaka

**Comment by eurekaka** [comment]
CI fails because https://github.com/pingcap/tidb/pull/21466 need to be merged first.

**Review by qw4990 — COMMENTED** [review]
LGTM

**Comment by qw4990** [comment]
PTAL @Reminiscent

**Review by Reminiscent — APPROVED** [review]
LGTM

**Review by qw4990 — APPROVED** [review]

**Pull request merged**